drivers/hmis/app/models/hmis/file.rb

```diff
@@ -35,11 +35,30 @@ def data_source
   scope :nonconfidential, -> { where(confidential: [false, nil]) }
 
   scope :viewable_by, ->(user) do
-    view_scope = where(client_id: Hmis::Hud::Client.with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files))
-    # view_scope = view_scope.nonconfidential unless user.can_view_any_confidential_client_files?
-    edit_scope = user.can_manage_own_client_files? ? where(user_id: user.id) : none
-
-    view_scope.or(edit_scope)
+    # NOTE: it's okay that confidential files are included in this scope even if the user
+    # doesn't have permission to read the file. Users can see the existence of confidential
+    # files but they can't read them. Reference:
+    # https://www.pivotaltracker.com/n/projects/2591838/stories/185293913
+    client_scope = Hmis::Hud::Client
+      .viewable_by(user)
+      .with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files)
+    enrollment_scope = Hmis::Hud::Enrollment
+      .viewable_by(user)
+      .with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files)
+
+    case_statement = Arel::Nodes::Case.new
+      .when(arel_table[:enrollment_id].not_eq(nil))
+      .then(arel_table[:enrollment_id].in(enrollment_scope.select(:id).arel))
+      .else(arel_table[:client_id].in(client_scope.select(:id).arel))
+
+    viewable_scope = Hmis::File
+      .left_outer_joins(:client)
+      .left_outer_joins(:enrollment)
+      .where(case_statement)
```

Comment:
So this is finding files where, if there's an enrollment_id, you have access to it, and if not, then it leans on the client access.  I _think_ that makes sense.  I haven't used `Arel::Nodes::Case.new` and am somewhat surprised to see it in a  where clause instead of a select, but it looks plausible.